File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/TrendyRepository.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * @author : simon
 * @description :
 * @last : 2024-11-03 17:31
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-03 17:31
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class TrendyRepository {
    private String author;
    private String name;
    private String avatar;
    private String url;
    private String description;
    private String language;
    private String languageColor;
    private Integer stars;
    private Integer forks;
    private Integer currentPeriodStars;
    private List<Contributor> builtBy;
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    @Builder
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Contributor {
        private String username;
        private String href;
        private String avatar;
    }
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/TrendyRepository.kt
package com.cpy3f2.gixor_mobile.model.entity

/**
 * @author : lya
 * @description :推荐的仓库
 * @last : 2024-11-03 17:31
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-03 17:31
 */
data class TrendyRepository(

    val name: String,
    val author: String,
    val avatar: String,
    val url: String?,
    val description: String?,
    val language: String?,
    val languageColor: String?,
    val stars: Int?,
    val forks: Int?,
    val currentPeriodStars: Int?,
    val builtBy: List<Contributor>?
)

data class Contributor(
    val username: String,
    val href: String,
    val avatar: String
)